.github/workflows/scripts/scanner-fetch-nvd-feeds.py

```diff
@@ -0,0 +1,143 @@
+        with gzip.open(gz_file_path, 'rb') as f_in:
+            with open(json_file_path, 'wb') as f_out:
+                shutil.copyfileobj(f_in, f_out)
+
+        # Optionally, remove the gzipped file after decompressing
+        os.remove(gz_file_path)
+        print(f"Saved {json_file_path}")
+
+    except Exception as e:
+        print(f"Error processing {url} for year {year}: {e}")
+        if os.path.exists(gz_file_path):
+            os.remove(gz_file_path)
+        if os.path.exists(json_file_path):
+            os.remove(json_file_path)
+
+def transform_json(input_file, output_file):
+    with open(input_file, 'r') as f_in, open(output_file, 'w') as f_out:
+        data = json.load(f_in)
+
+        for item in data['CVE_Items']:
+            impact_data = item.get('impact', {})
+            base_metric_v2 = impact_data.get('baseMetricV2', {})
+            base_metric_v3 = impact_data.get('baseMetricV3', {})
+
+            metrics = {}
+
+            if 'cvssV2' in base_metric_v2:
+                cvss_v2_data = base_metric_v2['cvssV2']
+                metrics["cvssMetricV2"] = [
+                    {
+                        "source": "nvd@nist.gov",
+                        "type": "Primary",
+                        "cvssData": {
+                            "version": "2.0",
+                            "vectorString": cvss_v2_data.get('vectorString', ''),
+                            "accessVector": cvss_v2_data.get('accessVector', ''),
+                            "accessComplexity": cvss_v2_data.get('accessComplexity', ''),
+                            "authentication": cvss_v2_data.get('authentication', ''),
+                            "confidentialityImpact": cvss_v2_data.get('confidentialityImpact', ''),
+                            "integrityImpact": cvss_v2_data.get('integrityImpact', ''),
+                            "availabilityImpact": cvss_v2_data.get('availabilityImpact', ''),
+                            "baseScore": cvss_v2_data.get('baseScore', 0)
+                        },
+                        "baseSeverity": base_metric_v2.get('severity', ''),
+                        "exploitabilityScore": base_metric_v2.get('exploitabilityScore', 0),
+                        "impactScore": base_metric_v2.get('impactScore', 0),
+                        "acInsufInfo": base_metric_v2.get('acInsufInfo', False),
+                        "obtainAllPrivilege": base_metric_v2.get('obtainAllPrivilege', False),
+                        "obtainUserPrivilege": base_metric_v2.get('obtainUserPrivilege', False),
+                        "obtainOtherPrivilege": base_metric_v2.get('obtainOtherPrivilege', False),
+                        "userInteractionRequired": base_metric_v2.get('userInteractionRequired', False)
+                    }
+                ]
+
+            if 'cvssV3' in base_metric_v3:
+                cvss_v3_data = base_metric_v3['cvssV3']
+                metrics["cvssMetricV31"] = [
```

Comment:
not sure how we can generate cvssMetricV31 vs cvssMetricV30 here as legacy feeds has the same field name **cvssV3**, is this a big concern?